SE_Topics.md

```diff
@@ -9,13 +9,17 @@ Please feel free to submit a PR with suggestions of topics to add.
 * Version control
 * Automated testing
 * Code review
-* Specification Writing
-* Code smells, naming practices, code style etc ?
+* Requirements Engineering
+* Code smells (perhaps merge with source code static analysis)
+* Naming practices, code style etc 
 * Bug reporting: how to write a good bug report ?
 * Pair programming
-
-
-
+* Software Modeling (how much should we model? using what? when (what kind of projects)? etc.)
```

Comment:
I see very little in the way of modelling used in practice, if I understand what you mean by modelling.  

Monitoring seems at least as relevant.

Reply:
I ran an internal (Google) survey of a few hundred engineers last month, and UML was literally the only proposed topic that had a majority that were *against* teaching it.

Reply:
The same experience for me. I had an undergrad student running a UML Cost Effectiveness survey with industry practitioners in California. The results are in the same direction.

However, the proposal is not to use UML necessarily but some sort of modeling. Or maybe not if the community converges to this point. 

I like the idea of modeling as sketch (https://www.martinfowler.com/bliki/UmlAsSketch.html) but not necessarily with UML unless the team feels comfortable and productive with it (as sketch).

There are some published papers that ran surveys on industry and pointed out informal modeling (without necessarily sticking to UML) has been seen as an important communication tool within team members.

Reply:
I think being able to sketch out some (informal) interaction diagrams is useful. We should just be careful about phrasing - "design diagram" might be less loaded than "modelling"?

Reply:
I agree.

Reply:
I suggest we collapse "Modelling" and "Requirements Engineering" into "Software Design"? 

I'll file an issue for myself to track down good examples / templates to work from.

Reply:
Hmm..I have doubts that putting RE into "Software Design" is bit of misleading. I never considered RE as part of designing software. Is there another way of structuring it?

Reply:
A reasonable design has to take the requirements and constraints as input.  
Maybe 
* Identifying requirements and constraints
* Software design

as two separate bullets?

Reply:
Sounds good to me. One may question the verb "identify" as limited because this may lead someone to think it is an isolated act of just identifying things and then moving to design whereas most of the times the requirements are constantly changing. So we are in many times adjusting requirements and constraints, analyzing conflicts, then tweaking the design, validating solutions, etc. That's why people created the term "Requirements Engineering" to involve activities around requirements stuff. But I agree that it might sound too prescriptive, overly complex, and not agile. I'm not against something simpler and more objective such as "Identifying/Managing/Changing requirements and constraints".

Reply:
That sounds great to me.